File: app/src/main/java/com/door43/translationstudio/ui/translate/OnReadModeListener.java
package com.door43.translationstudio.ui.translate;

public interface OnReadModeListener extends OnAdapterListener {
    void onOpenTargetTranslationCard(ReadModeAdapter.ViewHolder holder);
    void onCloseTargetTranslationCard(ReadModeAdapter.ViewHolder holder);
    CharSequence onRenderSourceText(ReadModeAdapter.ViewHolder holder);
    CharSequence onRenderTargetText(ReadModeAdapter.ViewHolder holder);
    void onOpenTranslationMode(String chapterSlug);
}

File: app/src/main/java/com/door43/translationstudio/ui/translate/OnReadModeListener.kt
package com.door43.translationstudio.ui.translate

import com.door43.translationstudio.rendering.model.TextNode

interface OnReadModeListener : OnAdapterListener {
    fun onOpenTargetTranslationCard(holder: ReadModeAdapter.ViewHolder)
    fun onCloseTargetTranslationCard(holder: ReadModeAdapter.ViewHolder)
    fun onRenderSourceText(holder: ReadModeAdapter.ViewHolder): List<TextNode>
    fun onRenderTargetText(holder: ReadModeAdapter.ViewHolder): List<TextNode>
    fun onOpenTranslationMode(chapterSlug: String)
}
